purchase_order_no_price/tests/test_purchase_order_no_price.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2025 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase
+
+
+class TestPurchaseOrderNoPrice(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.partner_id = cls.env["res.partner"].create({"name": "Test partner"})
```

Comment:
add the following checks:
* With a contact.
* self.assertTrue(self.purchase_id.order_line[0].price_unit) --> change it to a float compare with the specified price.
* self.assertFalse(self.purchase_id.order_line[1].price_unit) ---> change it to a float compare.